src/library.js

```diff
@@ -3303,8 +3303,13 @@ mergeInto(LibraryManager.library, {
     if (!func) {
       if (funcPtr >= wasmTableMirror.length) wasmTableMirror.length = funcPtr + 1;
       wasmTableMirror[funcPtr] = func = wasmTable.get(funcPtr);
+#if ASYNCIFY == 2
+      if (Asyncify.isAsyncExport(func)) {
+        wasmTableMirror[funcPtr] = func = Asyncify.makeAsyncFunction(func);
+      }
+#endif
     }
-#if ASSERTIONS
+#if ASSERTIONS && ASYNCIFY != 2 // With JSPI the function stored in the table will be a wrapper.
```

Comment:
Its it just the `wasmTableMirror` that contains that wrapper?  i.e. can this line now be reverted?

Reply:
This needs to remain. The `func` here could be `wasmTableMirror[funcPtr]` which is the wrapper so it won't be the same as the `wasmTable.get(funcPtr)` value.